SimplenoteUITests/NotesList.swift

```diff
@@ -6,12 +6,24 @@ class NoteList {
         app.tables.cells[noteName].tap()
     }
 
+    class func getNavBarIdentifier() -> String {
+        let navBar = app.navigationBars.element
+        guard navBar.exists else { return "" }
```

Comment:
Returning an empty string to signal that the nav bar wasn't found does the job, but could spiral into weird test failures downstream if a test doesn't check for the value not to be `""` and tries to access a nav bar with it.

A different take that leverages Swift's refined type system would be to use `Optional<String>`, `String?` for short. With this type, we can signal that no match was found by returning `.none`. The Swift compiler forces the consumer code to deal with the optionality, this is a great way to let future developers know that the nav bar may not have been found, without writing a single line of documentation.

The only changes required to the function are:

```suggestion
    class func getNavBarIdentifier() -> String? {
        let navBar = app.navigationBars.element
        guard navBar.exists else { return .none }
```

I haven't tried it, but I'm pretty sure the existing assertions won't need to change either.

Reply:
Thanks A LOT for this @mokagio! I will create a separate PR for this 🥳